src/Reflex/EventWriter.hs

```diff
@@ -0,0 +1,172 @@
+-- | This module provides 'EventWriterT', the standard implementation of
+-- 'EventWriter'.
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE Rank2Types #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+#ifdef USE_REFLEX_OPTIMIZER
+{-# OPTIONS_GHC -fplugin=Reflex.Optimizer #-}
+#endif
+module Reflex.EventWriter
+  ( EventWriterT (..)
+  , runEventWriterT
+  , EventWriter(..)
+  ) where
+
+import Reflex.Class
+import Reflex.Host.Class
+import Reflex.PerformEvent.Class
+import Reflex.PostBuild.Class
+import Reflex.TriggerEvent.Class
+
+import Control.Monad.Exception
+import Control.Monad.Identity
+import Control.Monad.Reader
+import Control.Monad.Ref
+import Control.Monad.State.Strict
+import Data.Dependent.Map (DMap, DSum (..))
+import qualified Data.Dependent.Map as DMap
+import Data.Functor.Misc
+import qualified Data.List.NonEmpty as NonEmpty
+import Data.Maybe
+import Data.Semigroup
+import Data.Some (Some)
+import qualified Data.Some as Some
+import Data.Tuple
+import Data.Unique.Tag
+
+-- | 'MonadEventWriter' efficiently collects 'Event' values using 'tellEvent'
+-- and combines them monoidally to provide an 'Event' result.
+class (Monad m, Monoid w) => EventWriter t w m | m -> t w where
```

Comment:
The class is named `EventWriter` but the comment refers to it as `MonadEventWriter`.